### PR TITLE
feat: move push sender to batch service

### PR DESCRIPTION
**Pull request by Ubisoft-potato**
Part of #405 

This PR will move push sender to batch service and delete old push sender code, manifest files.

Key changes:

- Add push metrics. [[1]](https://github.com/bucketeer-io/bucketeer/pull/995/files#diff-f73e5c9bbf677b202a36dfa0f9f9f7e7cef89cecd8e2d9ab46800de3ce0642a5R72-R80) [[2]](https://github.com/bucketeer-io/bucketeer/pull/995/files#diff-f73e5c9bbf677b202a36dfa0f9f9f7e7cef89cecd8e2d9ab46800de3ce0642a5R87)
- Push sender processor. [[1]](https://github.com/bucketeer-io/bucketeer/pull/995/files#diff-d04432152ef4b131259a7c41a96b4bbe74d34e8e8d24dfed546a82dd367c86fe)
- Register push sender. [[1]](https://github.com/bucketeer-io/bucketeer/pull/995/files#diff-9f3bffbf9b659d0228f7f25aeb18a14d9fa79a72e2a8100e7371b1d112faa00cR780-R789) [[2]](https://github.com/bucketeer-io/bucketeer/pull/995/files#diff-9f3bffbf9b659d0228f7f25aeb18a14d9fa79a72e2a8100e7371b1d112faa00cR308-R319)

### CODE REVIEW

File: pkg/batch/subscriber/processor/metrics.go [hunk 1]
```diff
@@ -68,11 +69,21 @@ var (
 			Name:      "subscriber_handled_event_total",
 			Help:      "Total number of handled messages",
 		}, []string{"subscriber", "code"})
+
+	pushHandledHistogram = prometheus.NewHistogramVec(
```
Comment (cre8ivejp) [inline pkg/batch/subscriber/processor/metrics.go:73]: How about creating a common histogram for the subscribers, like the `subscriberHandledCounter` and the `subscriberReceivedCounter` for future subscribers that use histogram?

File: manifests/bucketeer/charts/batch/values.yaml [hunk 2]
```diff
@@ -194,6 +195,15 @@ subscribers:
...
     workerNum: 1
+  pushSender:
+    project:
+    topic:
+    subscription:
+    pullerNumGoroutines: 5
+    pullerMaxOutstandingMessages: 1000
+    pullerMaxOutstandingBytes: 1000000000
+    maxMps: 50
+    workerNum: 1
```
Comment (cre8ivejp) [inline manifests/bucketeer/charts/batch/values.yaml:206]:
```suggestion
    pullerMaxOutstandingMessages: 500
    pullerMaxOutstandingBytes: 50000000
    maxMps: 100
    workerNum: 1
```

File: pkg/batch/subscriber/processor/metrics.go (hunk 1 above)
Comment (Ubisoft-potato) [reply pkg/batch/subscriber/processor/metrics.go:73]: Sure! 
It make sense, maybe there will be some other subscribers need to report this metric, for now, only push sender.

File: manifests/bucketeer/charts/batch/values.yaml (hunk 2 above)
Comment (Ubisoft-potato) [reply manifests/bucketeer/charts/batch/values.yaml]: Done!

File: pkg/batch/subscriber/processor/metrics.go (hunk 1 above)
Comment (Ubisoft-potato) [reply pkg/batch/subscriber/processor/metrics.go]: I have changed it's name to `subscriberHandledHistogram`, ref: https://github.com/bucketeer-io/bucketeer/pull/995/files#diff-d04432152ef4b131259a7c41a96b4bbe74d34e8e8d24dfed546a82dd367c86feR71-R74

File: pkg/batch/subscriber/processor/push_sender.go [hunk 3]
```diff
@@ -40,224 +38,132 @@ import (
...
 		return
 	}
 	msg.Ack()
-	handledCounter.WithLabelValues(codes.OK.String()).Inc()
+	subscriberHandledCounter.WithLabelValues(subscriberPushSender, codes.OK.String()).Inc()
 }
 
-func (s *sender) send(featureID, environmentNamespace string) error {
+func (p pushSender) send(featureID, environmentNamespace string) error {
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```
Comment (cre8ivejp) [inline pkg/batch/subscriber/processor/push_sender.go:127]: Could you please increase the timeout to 30 seconds?

Comment (Ubisoft-potato) [reply pkg/batch/subscriber/processor/push_sender.go:127]: Sure!

**Review by cre8ivejp — APPROVED** [review]
Thank you!

**Pull request merged**